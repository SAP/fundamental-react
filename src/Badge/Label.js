import classnames from 'classnames';
import { LABEL_TYPES } from '../utils/constants';
import PropTypes from 'prop-types';
import React from 'react';

const Label = ({ type, children, className, ...props }) => {
    const labelClasses = classnames(
        'fd-label',
        {
            [`fd-label--${type}`]: !!type
        },
        className
    );

    return <span {...props} className={labelClasses}>{children}</span>;
};
Label.displayName = 'Label';

Label.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    type: PropTypes.oneOf(LABEL_TYPES)
};

export default Label;
