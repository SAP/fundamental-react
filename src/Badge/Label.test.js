import Label from './Label';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<Label />', () => {
    const defaultLabel = <Label>Default</Label>;
    const typeLabel = (
        <Label className='blue' type='success'>
            Default
        </Label>
    );

    test('create label', () => {
        // create default label
        let component = renderer.create(defaultLabel);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // create success type label
        component = renderer.create(typeLabel);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the Label component', () => {
            const element = mount(<Label data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <Label ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('SPAN');
        expect(ref.current.className).toEqual('fd-label');
    });
});
