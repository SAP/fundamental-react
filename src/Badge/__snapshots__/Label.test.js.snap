// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> create label 1`] = `
<span
  className="fd-label"
>
  Default
</span>
`;

exports[`<Label /> create label 2`] = `
<span
  className="fd-label fd-label--success blue"
>
  Default
</span>
`;
